src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6680,23 +6693,26 @@ private StatementSyntax ParseStatementNoDeclaration(bool allowAnyExpression)
                     {
                         return this.ParseExpressionStatement(this.ParseQueryExpression(0));
                     }
-                    else
-                    {
-                        goto default;
-                    }
+                    break;
```

Comment:
I was thinking we could just add two cases above, `IsPossibleAwaitUsing`/`IsPossibleAwaitForeach`